src/app.js

```diff
@@ -497,7 +497,47 @@ var run = function () {
 
   var offsetToLineColumnConverter = new OffsetToLineColumnConverter(compiler.event)
 
-  var transactionDebugger = new Debugger('#debugger', editor, compiler, executionContext.event, switchToFile, offsetToLineColumnConverter)
+  var contentToolAPI = {
+    offsetToLineColumn: (location, file) => {
+      return offsetToLineColumnConverter.offsetToLineColumn(location, file, compiler.lastCompilationResult)
+    }
+  }
+
+  var ethToolAPI = {
+    toWei: (value, unit) => {
+      return executionContext.web3().toWei(value, unit)
+    }
+  }
+
+  var compilerAPI = {
+    lastCompilationResult: () => {
+      return compiler.lastCompilationResult
+    }
+  }
+
+  var editorAPI = {
```

Comment:
Please group by the component you pass this into, not by the component you take it from (and also adjust the naming to the nomenclature of the component you pass this object into).